ethcore/light/src/client/mod.rs

```diff
@@ -474,45 +474,20 @@ impl<T: ChainDataFetcher> Client<T> {
 	}
 
 	fn check_epoch_signal(&self, verified_header: &Header) -> Result<Option<Proof>, T::Error> {
-		use common_types::engines::machine::{AuxiliaryRequest, AuxiliaryData};
-
-		let mut block: Option<Vec<u8>> = None;
 		let mut receipts: Option<Vec<_>> = None;
 
 		loop {
 
 			let is_signal = {
```

Comment:
I don't like this `is_signal` variable, transition is more accurate. AFAIK, `signals_epoch_end` return either `Immediate transition` or `block signalling transition`.

Picking hair here but still.. maybe just match directly on `self.engine.signals_epoch_end(verified_header, receipts.as_ref().map(|x| &x[..]))`?

Reply:
> AFAIK, `signals_epoch_end` return either `Immediate transition` or `block signalling transition`.

It returns an [`EpochChange`](https://github.com/OpenEthereum/open-ethereum/blob/master/ethcore/engine/src/engine.rs#L135-L142) enum which is `Yes` (with proof), `No`, or `Unsure` (which used to hold an `AuxiliaryRequest` before this PR). Or do you mean something else?

I think using a local var is fine here, but happy to rename it. :)